tests/aws-cpp-sdk-dynamodb-integration-tests/TableOperationTest.cpp

```diff
@@ -459,6 +448,427 @@ TEST_F(TableOperationTest, TestThrottling)
     }
 }
 
+
+
+
+
+
+
+/*
+    after putItem requests have been made, it's not necessary that data is replicated to quorom before getItem is called.
+    
+    By default, getitem doesn't have strong consistency set. 
+    
+    So, it is possible for getItem to not have data available when called
+    
+    Hence, in such a situation, it is better to repeat the getItem for user sepcified times before evaluation
+
+    Using RetryPlanner class, we can break down a test suite into smaller parts and set the retry or exit plan accordingly
+
+    Idea is that a test suite will have a shareable context which will be available to all the blocks of the test suite
+
+    Like this, the preexisting test case can be made more robust in retrying parts which need retrying 
+
+    In this example, we break the tests into smaller blocks -> lambda functions which utilize context to update data needed for the test
+
+    It is important to note that because Gtest failure is sticky, a successful retry won't supersede the failure. Hence, it is important to 
+    evaluate using gtest macros when either tests pass or on the final retry.
+
+*/
+
+TEST_F(TableOperationTest, TestCrudOperations2)
+{
+    AWS_LOGSTREAM_TRACE(ALLOCATION_TAG, "TestCrudOperations")
+
+    struct TestContext{
+        Aws::Vector<PutItemOutcomeCallable> putItemResults;
+        Aws::Vector<GetItemOutcomeCallable> getItemOutcomes;
+        Aws::Vector<UpdateItemOutcomeCallable> updateItemOutcomes;
+        Aws::Vector<DeleteItemOutcomeCallable> deleteItemOutcomes;  
+    };
+
+    Aws::String crudTestTableName = BuildTableName(BASE_CRUD_TEST_TABLE);
+    const int numItems = 50;
+    CreateTable(crudTestTableName, numItems, numItems);
+
+    //now put 50 items in the table asynchronously
+    Aws::String testValueColumnName = "TestValue";
+   
+    auto crudTestTableNameSp = Aws::MakeShared<Aws::String>(ALLOCATION_TAG, crudTestTableName);
+
+
+    //Create context
+    auto contextSp = Aws::MakeShared< TestContext >  (
+        ALLOCATION_TAG
+    );
+
+    RetryPlanner<TestContext> retryObj(contextSp);
+
+    Aws::List<size_t> requestIds;
+    Aws::List<size_t> deleteRequestIds;
+
+    auto prepareRequests = [numItems, &requestIds](){
+        for(auto i = 0; i < numItems; ++i)
+        {
+            requestIds.push_back(i);
+        }
+    };
+    
+    prepareRequests();
+
+    deleteRequestIds = requestIds;
+
+    auto putItemReqFunction = [testValueColumnName, crudTestTableName, this, &requestIds](
```

Comment:
Example of a function block. Capture whatever is needed and adheres to the function signature of the test planner class